tide/master/rest/RestLogger.cpp

```diff
@@ -0,0 +1,84 @@
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer.                                                  */
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#include "RestLogger.h"
+#include <iostream>
+#include <string>
+
+#include <QJsonDocument>
+#include <QJsonObject>
+
+
+RestLogger::RestLogger( LoggingUtility& logger )
+    : _logger( logger )
+{}
+
+std::string RestLogger::getTypeName() const
+{
+    return _name;
+}
+
+std::string RestLogger::_toJSON() const
+{
+    const std::string& counterModificationTime = _logger.getCounterModificationTime();
+    const std::string& lastInteraction = _logger.getLastInteraction();
+    const std::string& lastInteractionTime = _logger.getLastInteractionTime();
+    const int windowCounter = _logger.getWindowCount();
+    const int windowActionCounter = _logger.getAccumulatedWindowCount();
+    const int interactionCounter = _logger.getRecentInteractionCounter();
+
+    std::string content;
+
+    QJsonObject obj;
+    QJsonObject event;
+    event["last_event"] = QString::fromStdString(lastInteraction) ;
```

Comment:
Replace std::string with QString in Logger and use the values here directly without conversion